app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -65,8 +80,38 @@ class AllFilesFragment : Fragment(), ItemClickListener {
         fetchFiles()
     }
 
-    override fun onItemClickListener() {
-        Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        val menuItem = menu.findItem(R.id.back_button)
+        menuItem.isVisible = viewModel.compareDirectories()
+    }
+
+    private fun updateOptionsMenu() {
+        requireActivity().invalidateOptionsMenu()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.back_button -> {
+                handleBackPress()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onFileClick(file: ListModel) {
+        updateOptionsMenu()
+        if (file.file.isDirectory) {
```

Comment:
can create a method in the ListModel class that would check if the file is a directory

**ListModel**
```
fun isDirectory(): Boolean {
    return file.isDirectory
}
```

**then call it here**

`if (file.isDirectory) {`